code/mission/missionparse.cpp

```diff
@@ -3445,8 +3445,13 @@ int parse_object(mission *pm, int  /*flag*/, p_object *p_objp)
 
 	// parse the persona index if present
 	// For backwards compatbility only
-	if (optional_string("+Persona Index:"))
+	if (optional_string("+Persona Index:")) {
 		stuff_int(&p_objp->persona_index);
+		if (p_objp->persona_index < -1 || p_objp->persona_index >= (int)Personas.size()) {
+			Warning(LOCATION, "Persona index for %s is out of range (%d)!  Setting to -1.", p_objp->name, p_objp->persona_index);
```

Comment:
This phrasing implies that the range is index. ` Persona index %d for %s` is clearer.

Reply:
Updated